src/main/java/core/basesyntax/service/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlersMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlersMap) {
+        this.operationHandlersMap = operationHandlersMap;
+    }
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation operation) {
+        if (operationHandlersMap.containsKey(operation)) {
+            return operationHandlersMap.get(operation);
+        }
+        throw new RuntimeException("Operation " + operation + " not supported");

```

Comment:
Don't forget to fix tests
```suggestion
        throw new IllegalArgumentException("Operation " + operation + " not supported");
```